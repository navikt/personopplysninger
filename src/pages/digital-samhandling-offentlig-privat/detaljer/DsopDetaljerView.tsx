import ListElement from '@/components/listelement/ListElement';
import { FormattedMessage } from 'react-intl';
import FileSaver from 'file-saver';
import { DsopInfo } from '@/types/dsop';
import { BodyShort, Button, Heading } from '@navikt/ds-react';
import dayjs from 'dayjs';
import { Liste } from '@/components/listelement/Liste';

interface Props {
    id: string;
    dsopInfo: DsopInfo;
}

const DsopDetaljerView = (props: Props) => {
    const { id, dsopInfo } = props;
    const innslag = dsopInfo.filter((d) => d.uthentingsTidspunkt === id).shift();

    return innslag ? (
        (() => {
            const onClick = () => {
                const leverteData = JSON.parse(atob(innslag.leverteData));
                const fileContent = JSON.stringify(leverteData, null, 2);
                const fileBlob = new Blob([fileContent], {
                    type: 'application/json',
                });
                const fileName = 'utleverte-data.json';
                FileSaver.saveAs(fileBlob, fileName);
            };
            return (
                <div>
                    <div className="detaljer__tittel">
                        <Heading level="2" size="small">
                            {innslag.mottakernavn}
                        </Heading>
                        <BodyShort>
                            <FormattedMessage id="side.organisasjonsnummer" values={{ orgnr: innslag.mottaker }} />
                        </BodyShort>
                    </div>
                    <hr className="box__linje-bred" />
                    <div className="box">
                        <div className="box__content">
                            <Liste>
                                <ListElement
                                    titleId={'dsop.uthentingstidspunkt'}
                                    content={dayjs(innslag.uthentingsTidspunkt).format('DD.MM.YYYY hh:mm')}
                                />
                                <ListElement titleId={'dsop.personnummer'} content={innslag.person} />
                                <ListElement titleId={'dsop.tema'} content={innslag.tema} />
                            </Liste>
                        </div>
                        <div className="detaljer__container">
                            <Button variant="primary" onClick={onClick}>
                                <FormattedMessage id="dsop.lastned" />
                            </Button>
                        </div>
                    </div>
                </div>
            );
        })()
    ) : (
        <div>
            <FormattedMessage id="dsop.ingendata" />
        </div>
    );
};

export default DsopDetaljerView;
