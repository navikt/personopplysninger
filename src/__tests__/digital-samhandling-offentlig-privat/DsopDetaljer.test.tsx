import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import DsopDetaljerView from '@/pages/digital-samhandling-offentlig-privat/detaljer/DsopDetaljerView';
import dsopInfo from '@/clients/apiMock/app/fetch/dsop-info.json';
import nbMessages from '@/text/nb';

describe('DsopDetaljerView', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <IntlProvider locale={'nb'} messages={nbMessages}>
                <DsopDetaljerView dsopInfo={dsopInfo} id={'2019-06-18T10:45:51.634'} />
            </IntlProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
