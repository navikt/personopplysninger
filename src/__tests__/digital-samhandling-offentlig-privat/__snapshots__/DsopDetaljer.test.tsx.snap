// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DsopDetaljerView renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="detaljer__tittel"
    >
      <h2
        class="navds-heading navds-heading--small"
      >
        STOREBRAND LIVSFORSIKRING AS
      </h2>
      <p
        class="navds-body-short navds-body-short--medium"
      >
        Organisasjonsnummer 958995369
      </p>
    </div>
    <hr
      class="box__linje-bred"
    />
    <div
      class="box"
    >
      <div
        class="box__content"
      >
        <dl
          class="list"
        >
          <div
            class="list-item"
          >
            <dt
              class="list-element-heading"
            >
              <span
                class="navds-label"
              >
                Uthentingstidspunkt
              </span>
            </dt>
            <dd>
              18.06.2019 10:45
            </dd>
          </div>
          <div
            class="list-item"
          >
            <dt
              class="list-element-heading"
            >
              <span
                class="navds-label"
              >
                Personnummer
              </span>
            </dt>
            <dd>
              04019727454
            </dd>
          </div>
          <div
            class="list-item"
          >
            <dt
              class="list-element-heading"
            >
              <span
                class="navds-label"
              >
                Tema
              </span>
            </dt>
            <dd>
              AAP
            </dd>
          </div>
          <div
            class="list-item"
          >
            <dt
              class="list-element-heading"
            >
              <span
                class="navds-label"
              >
                Behandlingsgrunnlag
              </span>
            </dt>
            <dd>
              Samtykke
            </dd>
          </div>
        </dl>
      </div>
      <div
        class="detaljer__container"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
        >
          <span
            class="navds-label"
          >
            Last ned leverte data
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
